garak/resources/gcg/generate_gcg.py

```diff
@@ -154,13 +162,23 @@ def generate_gcg(
         logfile = kwargs["logfile"]
     else:
         timestamp = datetime.now().strftime("%Y%m%dT%H%M%S")
-        model_string = "_".join([x.replace("/", "-") for x in model_names])
+        if target_generator is not None:
+            model_string = target_generator.name
+        elif model_names is not None:
+            model_string = "_".join([x.replace("/", "-") for x in model_names])
+        else:
+            msg = "You must specify either a target generator or a list of model names to run GCG!"
+            logger.error(msg)
+            raise Exception(msg)
         logfile = (
             garak._config.transient.basedir
-            / "resources/gcg/data/logs/{timestamp}_{model_string}.json"
+            / f"resources/gcg/data/logs/{timestamp}_{model_string}.json"
```

Comment:
Is `_config.run_id` available when this is executed?

Reply:
Worth checking. It should be, unless using it as a library. I can check for it and if it's not there, fall back to timestamp.